urlpattern/resources/urlpatterntestdata.json

```diff
@@ -2367,15 +2375,24 @@
   },
   {
     "pattern": [{ "hostname": "bad#hostname" }],
```

Comment:
Could you clarify why only `#%/` don't throw errors and other hostname patterns like "bad\\:hostname" or "bad>hostname" continue throw errors?

btw I noticed "bad\\:hostname" updates hostname in URL.

```
u = new URL('http:/dummy.site')
u.hostname = "bad\\:hostname"
u.hostname  // bad
```

Reply:
Host parser (specifically domain to ASCII with domain and false) strip all trailing values whenever it sees # https://url.spec.whatwg.org/#concept-host-parser

Reply:
The host parser does not do that. Do you mean the `host` setter or some such?

Reply:
> The host parser does not do that. Do you mean the `host` setter or some such?

Yes, I meant the host setter, which calls the host parser.

> Could you clarify why only #%/ don't throw errors and other hostname patterns like "bad\:hostname" or "bad>hostname" continue throw errors?

@sisidovski Sorry I missed this comment. URL spec hostname state step 3 (ref: https://url.spec.whatwg.org/#hostname-state) says that...

if c is the EOF code point, U+002F (/), U+003F (?), or U+0023 (#) ... then decrease pointer by 1, and then: ... Let host be the result of host parsing buffer with url is not special.

So, if you see these characters, we take the existing non-empty buffer and parse it as a valid hostname.